### PR TITLE
Basic TypeORM integration

### PR DESCRIPTION
**Pull request by mkalam-alami**
Making a pull request to review the code more comfortably (this is mostly 4 month old, quite sensitive code that had yet to be merged)

Also includes:

- #426 Alert API, with integration on a couple first forms as a test
- Some work on TS typings
- A fix on server startup (at the first launch only)

### CODE REVIEW

File: client/scripts/common/alerts.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Bootstrap Notify integration
+ * https://github.com/mouse0270/bootstrap-notify
+ */
+
+interface Alert {
+  type: "success" | "info" | "warning" | "danger";
```
Comment (Aurel300) [inline client/scripts/common/alerts.ts]: [String enum](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums)?

File: client/scripts/common/alerts.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  title?: string;
+  message: string;
+  floating: boolean;
+}
+
+const INLINE_ALERTS_SELECTOR = "#js-alerts-inline";
+
+function makeInlineAlertsDismissable() {
+  const inlineAlerts = $(".alert", $(INLINE_ALERTS_SELECTOR));
+  inlineAlerts.css({ cursor: "pointer "});
```
Comment (Aurel300) [inline client/scripts/common/alerts.ts]: Quote, then space.

File: package.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
     "start:production": "node dist/server/index.js",
     "build": "npm run build:css && npm run build:client && npm run build:server",
     "build:css": "node -r ts-node/register server/sass.ts",
+    "build:sass": "echo 'Did you mean build:css?'",
```
Comment (Aurel300) [inline package.json]: Could just run the `build:css` script.

File: server/entity/user.entity.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  public email: string;
+
+  @Column({ nullable: true })
+  public avatar: string;
+
+  @Column({ default: false })
+  public is_mod: string; // XXX "true" or ""
+
+  @Column({ default: false })
+  public is_admin: string; // XXX "true" or ""
```
Comment (Aurel300) [inline server/entity/user.entity.ts]: These *really* should be migrated into bools :)

File: package.json (hunk 3 above)
Comment (mkalam-alami) [reply package.json]: I did at first bur realized stuff like "npm run build:sass -- --watch" didn't work as expected (the argument wasn't pushed down properly). Not sure if there's a way to pass npm arguments as is.

File: server/entity/user.entity.ts (hunk 4 above)
Comment (mkalam-alami) [reply server/entity/user.entity.ts]: I reviewed it during my own CR and it's actually even worse than that! https://github.com/alakajam-team/alakajam/blob/master/server/entity/user.entity.ts#L33

File: package.json (hunk 3 above)
Comment (Aurel300) [reply package.json]: I meant literally to copy the command for `build:css`. Duplication, but it works.

**Comment by mkalam-alami** [comment]
> I find the JSON-pushed-to-HTML for alerts thing a bit hacky, but I guess it works.

It's strange indeed. It's not the first time I used that approach on the site honestly, ie. sharing state from server to client through dumped JSON. With the architecture we have (server-side rendering with *some* client logic), it's strangely the best approach I found (so far).

> The types for alerts might be nicer with discriminated unions also, i.e. define a separate "type" for each alert that might happen, then union them. Severity can also be discriminated based on this type.

I think discriminated unions is only useful is properties differ depending on the type right? (if yes it would not apply here due to the props being identical whatever the type)

**Comment by Aurel300** [comment]
> I think discriminated unions is only useful is properties differ depending on the type right? (if yes it would not apply here due to the props being identical whatever the type)

Right, but my point is that the props should not necessarily be identical. E.g. instead of:

 - `"danger" alert with message "bad login"`
 - `"danger" alert with message "unauthorised access to /mod-tools"`
 - `"success" alert with message "logged in"`

We could have something like

 - `BadLogin`
 - `Unauthorised("/mod-tools")`
 - `LoginOk`

The message and colour/severity of an alert can be generated in most cases from just its `kind` property. If there is any extra information, e.g. the hypothetical `Unauthorised` alert, it is an extra property of that type.

**Pull request merged**